GameServer.cpp

```diff
@@ -460,33 +460,27 @@ void GameServer::DoTimedUpdates()
 	LogCounters(counters);
 }
 
-std::size_t GameServer::FindGameInfoClient(sockaddr_in &from, unsigned int clientRandValue)
+std::size_t GameServer::FindGameInfoClient(const sockaddr_in &from, unsigned int clientRandValue)
 {
-	// Search games list
-	for (std::size_t i = 0; i < numGames; ++i)
-	{
-		// Check if this address matches
-		if ((gameInfo[i].clientRandValue == clientRandValue) && (memcmp(&gameInfo[i].addr, &from, sizeof(sockaddr_in)) == 0))
-		{
-			// Return the GameInfo
-			return i;
-		}
-	}
-
-	// GameInfo not found
-	return InvalidGameInfoIndex;
+	return FindGameInfo(from,
+		[clientRandValue](const GameInfo& gameInfo) {return gameInfo.serverRandValue == clientRandValue; }
+	);
 }
 
+std::size_t GameServer::FindGameInfoServer(const sockaddr_in &from, unsigned int serverRandValue)
+{
+	return FindGameInfo(from, 
+		[serverRandValue](const GameInfo& gameInfo) {return gameInfo.serverRandValue == serverRandValue; }
+	);
+}
 
-std::size_t GameServer::FindGameInfoServer(sockaddr_in &from, unsigned int serverRandValue)
+std::size_t GameServer::FindGameInfo(const sockaddr_in& from, const std::function <bool(const GameInfo&)>& compareFunction)
 {
-	// Search games list
 	for (std::size_t i = 0; i < numGames; ++i)
 	{
 		// Check if this address matches
-		if ((gameInfo[i].serverRandValue == serverRandValue) && (memcmp(&gameInfo[i].addr, &from, sizeof(sockaddr_in)) == 0))
+		if (compareFunction(gameInfo[i]) && (memcmp(&gameInfo[i].addr, &from, sizeof(sockaddr_in)) == 0))

```

Comment:
I wasn't sure about passing gameInfo in this manner, since it is actually a pointer. My goal was to pass a reference to the indexed value. I'm not sure how to exactly test this rewritten function as it isn't called if I just start NetFixServer on my machine.